qoajad_backend/src/main/java/com/qoajad/backend/database/DatabaseAccessorImplementation.java

```diff
@@ -0,0 +1,89 @@
+package com.qoajad.backend.database;
+
+import com.qoajad.backend.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@Qualifier(value = "defaultDatabaseAccessor")
+public class DatabaseAccessorImplementation implements DatabaseAccessor {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public DatabaseAccessorImplementation(final JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+        Objects.requireNonNull(jdbcTemplate, "To instantiate the database accessor the jdbc template must not be null.");
+    }
+
+    @Override
+    public List<User> retrieveAllUsers() {
+        List<User> users;
+        try {
+            final String query = "SELECT user_id, user_pw FROM User";
+            users = jdbcTemplate.query(query, (resultSet, rowNum) -> {
+                final int userId = resultSet.getInt("user_id");
+                final String password = resultSet.getString("user_pw");
+                return new User(userId, password);
+            });
+        } catch (Exception e) {
+            throw e;
+        }
+        return users;
```

Comment:
It's useful to avoid having nulls in the code whenever possible.

```suggestion
        return users == null ? Collections.emptyList() : users;
```